mozci/util/hgmo.py

```diff
@@ -25,7 +26,8 @@ class HGMO:
     )
 
     # instance cache
-    CACHE: Dict[Tuple[str, str], HGMO] = {}
+    CACHE: Dict[Tuple[str, str], HgRev] = LRU(1000)
+    JSON_PUSHES_CACHE: Dict[int, Dict[str, Any]] = LRU(1000)
```

Comment:
Nit: You could define a Push object like: `HgPush = NewType("HgPush", Dict[str, Any])`